client/my-sites/drafts/draft-list.jsx

```diff
@@ -152,4 +183,14 @@ var Drafts = React.createClass( {
 	}
 } );
 
-module.exports = DraftList;
+module.exports = connect( ( state, { siteID } ) => {
+	const counts = {
+		me: getNormalizedMyPostCounts( state, siteID, 'post' ).draft,
+		everyone: getNormalizedPostCounts( state, siteID, 'post' ).draft
+	};
+
+	return {
+		isRequestingCounts: isRequestingPostCounts( state, siteID, 'post' ),
+		counts
```

Comment:
You're counteracting the benefits of pure render by creating a new `counts` object every time this selector is run. Because it's a new object, shallow comparison will always flag it as changed. You should pass simple props instead (i.e. `myCounts`, `everyoneCounts`).